### PR TITLE
Pb99u069/887 drop proxy

### LINKED ISSUE #887: Drop `Proxy` parameter from `mkCallType`/`mkOgmiosCallType`...
Given most (all?) uses these functions appear with a top-level type-hint, the `Proxy` is superfluous

### PR DESCRIPTION
**Pull request by pb99u069**
Closes #887

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -90,6 +90,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
   - The config accepted by `launchCtlRuntime` and `buildCtlRuntime` now takes a `ctl-server.enable` field. If `false`, `ctl-server` will not be launched.
 - `SlotLength` and `RelativeTime` in `EraSummary` from Ogmios are now of type `Number` instead of `BigInt`. Also add `Maybe` around some functions in `Type.Interval` or changed it's signature to use `Number`. ([#868](https://github.com/Plutonomicon/cardano-transaction-lib/issues/868))
 - Renamed `UtxoM` to `UtxoMap` ([#963](https://github.com/Plutonomicon/cardano-transaction-lib/pull/963))
+- Removed the superfluous `Proxy` parameter from `mkCallType` and `mkOgmiosCallType`. ([#887](https://github.com/Plutonomicon/cardano-transaction-lib/issues/887))
```
Comment (klntsky) [inline CHANGELOG.md:93]: These are internal changes, no need for a changelog entry.

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/QueryM/JsonWsp.purs [hunk 2]
```diff
@@ -76,30 +75,29 @@ type JsonWspResponse (a :: Type) =
   }
 
 -- | A wrapper for tying arguments and response types to request building.
-newtype JsonWspCall :: Type -> Type -> Type
-newtype JsonWspCall (i :: Type) (o :: Type) = JsonWspCall
```
Comment (klntsky) [inline src/QueryM/JsonWsp.purs:80]: I think the task was to remove the `Proxy` parameter from the functions and not the type parameter from `JsonWspCall`

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/QueryM/JsonWsp.purs (hunk 2 above)
Comment (ngua) [reply src/QueryM/JsonWsp.purs:80]: That's right, we definitely want to keep the type parameter here

File: src/QueryM/Ogmios.purs [hunk 3]
```diff
@@ -152,59 +151,59 @@ import Untagged.Union (type (|+|), toEither1)
 --------------------------------------------------------------------------------
 
 -- | Queries Ogmios for the system start Datetime
-querySystemStartCall :: JsonWspCall Unit SystemStart
-querySystemStartCall = mkOgmiosCallType Proxy
+querySystemStartCall :: JsonWspCall Unit
```
Comment (ngua) [inline src/QueryM/Ogmios.purs:154]: Removing the type parameter leads to less meaningful/informative type signatures for the different calls

**Review by ngua — CHANGES_REQUESTED** [review]

File: src/QueryM/JsonWsp.purs (hunk 2 above)
Comment (pb99u069) [reply src/QueryM/JsonWsp.purs]: sorry, I struggled to understand the idea of `Proxy`. hope it's ok now

**Comment by pb99u069** [comment]
I didn't get errors with `spago build` in this branch, but haven't checked the tests. I'll do that

**Review by klntsky — APPROVED** [review]

**Pull request merged**